### ISSUE TITLE
DAGP sometimes resolves phantom dependencies

### ISSUE THREAD

**Issue by ZacSweers** [autonomousapps/dependency-analysis-gradle-plugin#981]
**Build scan link**

**Plugin version**
1.24.0

**Gradle version**
8.3

**JDK version**
20

**(Optional) Android Gradle Plugin (AGP) version**
8.3.0-alpha06

**(Optional) `reason` output for bugs relating to incorrect advice**

**Describe the bug**

There are a few dependencies that DAGP weirdly reports phantom replacements for. One is with androidx.lifecycle dependencies. The below snippet is from `final-advice.json`

```json
      "coordinates": {
        "type": "module",
        "identifier": "androidx.lifecycle:lifecycle",
        "resolvedVersion": "2.7.0-alpha02",
        "gradleVariantIdentification": {
          "capabilities": [
            "androidx.lifecycle:lifecycle"
          ],
          "attributes": {
            "org.jetbrains.kotlin.platform.type": "jvm"
          },
          "externalVariant": {
            "capabilities": [
              "androidx.lifecycle:lifecycle-common"
            ],
            "attributes": {
              "org.jetbrains.kotlin.platform.type": "jvm"
            }
          }
        }
      },
      "toConfiguration": "api"
    },
```

The problem here is that `androidx.lifecycle:lifecycle` doesn't actually exist. `androidx.lifecycle:lifecycle-common` is the correct dependency, but listed as a capability 🤔.

Another example is `com.squareup.leakcanary:leakcanary-object`, resolved from `com.squareup.leakcanary:leakcanary-object-watcher-android`

**To Reproduce**
1. Checkout this repo: https://github.com/zacsweers/catchup
2. Run `./scripts/rake_dependencies.sh`
3. Look in `app/build/reports/dependency-analysis/final-advice.json`

**Expected behavior**

It shouldn't report phantom dependencies

**Additional context**

I noticed this in both 1.24.0 and previous versions, so I don't think it's new with the KMP changes.

**Comment by ZacSweers** [comment]
CC @jjohannes

**Comment by jjohannes** [comment]
I suspect this to be somehow caused by the changes in #919. This looks similar to https://github.com/autonomousapps/dependency-analysis-gradle-plugin/issues/963#issuecomment-1750748868

I haven't looked at the code of #919 in detail, but could it be that the `"org.jetbrains.kotlin.platform.type": "jvm"` somehow makes the plugin think this is an external KPM variant and then "assumes" that  `androidx.lifecycle:lifecycle` has to exist as the "main variant" name.

This part in the identifier is indicating this:
```
          "externalVariant": {
            "capabilities": [
              "androidx.lifecycle:lifecycle-common"
            ],
            "attributes": {
              "org.jetbrains.kotlin.platform.type": "jvm"
            }
          }
```

`externalVariant` did not exist before #919. 

That there is also...

```
"capabilities": [
  "androidx.lifecycle:lifecycle"
],
```

...looks like an undesired side effect of the new functionality. I don't thing this issue is related to the _capability support_.

**Comment by autonomousapps** [comment]
Closing as probably resolved with v1.26.0.
